src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -47,39 +47,29 @@ std::unique_ptr<CascadiaSettings> CascadiaSettings::LoadAll(const bool saveOnLoa
     {
         const auto actualData = fileData.value();
 
-        JsonValue root{ nullptr };
-        bool parsedSuccessfully = JsonValue::TryParse(actualData, root);
-        // TODO:MSFT:20737698 - Display an error if we failed to parse settings
-        if (parsedSuccessfully)
+        // If Parse fails, it'll throw a hresult_error
+        JsonObject root = JsonObject::Parse(actualData);
+
+        resultPtr = FromJson(root);
+
+        //  Update profile only if it has changed.
+        if (saveOnLoad)
         {
-            JsonObject obj = root.GetObjectW();
-            resultPtr = FromJson(obj);
+            const JsonObject json = resultPtr->ToJson();
+            auto serializedSettings = json.Stringify();
 
-            //  Update profile only if it has changed.
-            if (saveOnLoad)
+            if (actualData != serializedSettings)
             {
-                const JsonObject json = resultPtr->ToJson();
-                auto serializedSettings = json.Stringify();
-
-                if (actualData != serializedSettings)
-                {
-                    resultPtr->SaveAll();
-                }
+                resultPtr->SaveAll();

```

Comment:
I'm only 50% on whether we should be writing the settings out on load honestly

Reply:
As a use case for writing back out - people would need to delete their profiles.json file to get the app to write out the file including keybindings once that PR is in. Otherwise they'd have to add all the keybindings manually.

Reply:
That's true ... but also, we might want to document it as a stopgap that is only necessary while we don't have a settings UI or some sort of "save my settings" functionality

Reply:
An alternative solution might be to to write an example profile to the directory without modifying the user's profile.

Another alternative might be that once a json library is in use that supports comments, properties that haven't changed from the current default value could be saved commented, while ones explicitly changed by the user would actually be written out.

Regardless though, always modifying the configuration file when unnecessary is unfortunate.

Reply:
I like the idea of dumping an example.

Reply:
This seems like a discussion that's maybe better off in #754